src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeBodyPartRotationalData.java

```diff
@@ -0,0 +1,157 @@
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeMapValue;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class ImmutableSpongeBodyPartRotationalData extends AbstractImmutableData<ImmutableBodyPartRotationalData, BodyPartRotationalData> 
+    implements ImmutableBodyPartRotationalData {
+
+    private final Map<BodyPart, Vector3d> rotations;
+    private final ImmutableMapValue<BodyPart, Vector3d> rotationsValue;
+    private final ImmutableValue<Vector3d> headRotationValue;
+    private final ImmutableValue<Vector3d> chestRotationValue;
+    private final ImmutableValue<Vector3d> leftArmRotationValue;
+    private final ImmutableValue<Vector3d> rightArmRotationValue;
+    private final ImmutableValue<Vector3d> leftLegRotationValue;
+    private final ImmutableValue<Vector3d> rightLegRotationValue;
+
+    public ImmutableSpongeBodyPartRotationalData(Map<BodyPart, Vector3d> rotations) {
+        super(ImmutableBodyPartRotationalData.class);
+        
+        this.rotations = ImmutableMap.copyOf(checkNotNull(rotations));
+        this.rotationsValue = new ImmutableSpongeMapValue<>(Keys.BODY_ROTATIONS, this.rotations);
+        this.headRotationValue = new ImmutableSpongeValue<>(Keys.HEAD_ROTATION, Optional.ofNullable(this.rotations.get(BodyParts.HEAD)).orElse(DataConstants.DEFAULT_HEAD_ROTATION));
+        this.chestRotationValue = new ImmutableSpongeValue<>(Keys.CHEST_ROTATION, Optional.ofNullable(this.rotations.get(BodyParts.CHEST)).orElse(DataConstants.DEFAULT_CHEST_ROTATION));
+        this.leftArmRotationValue = new ImmutableSpongeValue<>(Keys.LEFT_ARM_ROTATION, Optional.ofNullable(this.rotations.get(BodyParts.LEFT_ARM)).orElse(DataConstants.DEFAULT_LEFT_ARM_ROTATION));
+        this.rightArmRotationValue = new ImmutableSpongeValue<>(Keys.RIGHT_ARM_ROTATION, Optional.ofNullable(this.rotations.get(BodyParts.RIGHT_ARM)).orElse(DataConstants.DEFAULT_RIGHT_ARM_ROTATION));
+        this.leftLegRotationValue = new ImmutableSpongeValue<>(Keys.LEFT_LEG_ROTATION, Optional.ofNullable(this.rotations.get(BodyParts.LEFT_LEG)).orElse(DataConstants.DEFAULT_LEFT_LEG_ROTATION));
+        this.rightLegRotationValue = new ImmutableSpongeValue<>(Keys.RIGHT_LEG_ROTATION, Optional.ofNullable(this.rotations.get(BodyParts.RIGHT_LEG)).orElse(DataConstants.DEFAULT_RIGHT_LEG_ROTATION));
+        
+        registerGetters();
+    }
+
+    @Override
+    public BodyPartRotationalData asMutable() {
+        return new SpongeBodyPartRotationalData(this.rotations);
+    }
+
+    @Override
+    public int compareTo(ImmutableBodyPartRotationalData o) {
+        return 0;
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        return new MemoryDataContainer()
+                .set(Keys.BODY_ROTATIONS, this.rotations)
+                .set(Keys.HEAD_ROTATION, this.rotations.get(BodyParts.HEAD))
```

Comment:
I think each Vector3d should be broken down into it's x y and z parts @gabizou confirm?